### PR TITLE
Inline Help: Stop browsing to broken route on enter keypress

### PR DESCRIPTION
**Pull request by jblz**
Currently, if I press <kbd>enter</kbd> when the search field has focus, I end up at an invalid route.

## To Reproduce

* Open inline help
* Hit enter while the input field has focue (with or without entering text first)
* Get redirected to https://wordpress.com/settings/writing/false

## In this change:
* Refactor redux for simpler href access via `getInlineHelpCurrentlySelectedLink` selector
* Move `openResult` to the top level component
* Pass `openResult` to the search card & results components
* Pass the href from the child components into the provided `openResult` fn
* Use `lodash#get` throughout the selector to guard against fatals
* Prettify along the way

## To Test

* Run this branch
* Open inline help
* Click in the search box
* Hit the enter key (with and without text present in the field)
  * It should not redirect to an invalid route & searching should work as before
  * No `calypso_inlinehelp_link_open` events should be triggered
* Press the up and down arrow keys to "select" a result
* Hit the enter key
  * It should redirect to the selected support article
  * A properly-formed `calypso_inlinehelp_link_open` events should be triggered
* Go back to calypso and open inline help
* Clicking a result (with and without having searched or selected an item) should trigger a properly-formed `calypso_inlinehelp_link_open` event & redirect to the support article

### CODE REVIEW

File: client/state/inline-help/selectors.js [hunk 1]
```diff
@@ -1,31 +1,27 @@
...
 }
 
 /**
  * Returns the index of the currently selected search result.
  *
  * @param  {Object}  state  Global state tree
- * @return {String}        The index of the currently selected search result
+ * @return {Integer}        The index of the currently selected search result
  */
 export function getSelectedResult( state ) {
```
Comment (lsinger) [inline client/state/inline-help/selectors.js]: Looking at this now I believe we should rename this to `getSelectedResultIndex`. Thoughts?

File: client/state/inline-help/selectors.js [hunk 2]
```diff
@@ -49,9 +45,21 @@ export function isRequestingInlineHelpSearchResultsForQuery( state, searchQuery
...
 	}
 	return searchResults;
 }
+
+/**
+ * Returns the link / href of the selected search result item
+ * @param  {Object}  state  Global state tree
+ * @return {String}         The href of the selected link target
+ */
+export function getInlineHelpCurrentlySelectedLink( state ) {
```
Comment (lsinger) [inline client/state/inline-help/selectors.js]: If we rename `getSelectedResult` to `getSelectedResultIndex`, this could make sense as `getSelectedResult`?

**Comment by lsinger** [comment]
Works as advertised. Thanks for fixing this! 

I left two naming-related comments, but I don't consider those to be critical. Feel free to merge at your discretion.

**Comment by jblz** [comment]
Thanks for the review, @lsinger. I'm pretty sure I agree w/ the naming changes you recommended, but I think I'll go ahead and merge this & change those separately.

**Pull request merged**